### ISSUE TITLE
The Language is incredibly scuffed and inefficient (obviously). 

### ISSUE THREAD

**Issue by KittyCatOfMagic27** [KittyCatOfMagic27/ADPL#1]
When I started this project I thought that I could just compile directly to NASM like Tsoding did (big mistake). As I was writing it I was realizing more and more how bad of an idea that was since he was making a stack-based language and I wasn't. It steadily got bigger and bigger, and then I decided that all of it was just too much hassel in the way I was doing it. I already refatored the whole thing once, but I still clung onto my idea of nodes. I had a lot of pride with that, needless to say, _bad idea_. SO! Here we are with a funky little thing is getting progressively harder to work in. 

   My fix? Well I looked again at guides and what other people have done, and I decided to scrap the node idea and compile it into an intermediate language. Which one? Welp, I'm going to be stubbord again and make my own :D. Doing this allows me customize it to what I need and want even though I'm sure other people have made amazing intermediate languages I can just use. I wont make it too much to do though, it will just be a more readable version of what the NASM does. For example, at the time of working on this defining a variable in this language is  something like this:
def x 32BIT 69;

  This is the current plan and I may put an update here if something else changes, but for now 'cya!

**Comment by KittyCatOfMagic27** [comment]
UPDATE:
   Got the syntax for the intermediate language down pat! And it is very helpful in development since it allows everything to be broken down in to smaller blocks for the compiler to deal with!!! I've reimplmented variables nearly completely already, all that's left to do on that front is strings. Functions are a lot easier to deal with in development and will hopefully be less janky and will support more arguments and be able to return values!  
This is the syntax as of current:
`Label: foo ; 
arg w 8BIT ; 
end ; 
Label: _start ; 
def x 32BIT 69 ; 
def y 32BIT 420 ; 
def z 32BIT ; 
z = + x y ; 
x = * y x ; 
x = / x y ; 
x = % x y ; 
x = - y x ; 
in true ; 
call foo ; 
end ; `

Currently working on function calls btw. Anyway, have a nice day! ^^

**Comment by KittyCatOfMagic27** [comment]
UPDATE:
Got rid of parentheses around variables that are assigned/reassigned to expressions and the write to symbol for reassignment. Loops have been reimplemented aswell, will start working on ifs, elifs, and elses tomorrow. 
Just to show the new syntax and stuff here is a example that prints the nth fibonacci number, _yknow basic loop demo stuff_:
`fn main()   
  let n 5   
  n--    
  let x 1    
  let y 1    
  let i 0    
  loop(i<n)    
    let z x + y    
    x y    
    y z    
    i++   
  end   
  write(x)   
end`
And here is it compiled into ADIL _(A Dumb Intermediate Language)_:
`Label: _start ; 
def n 32BIT 5 ; 
-- n ; 
def x 32BIT 1 ; 
def y 32BIT 1 ; 
def i 32BIT 0 ; 
Label: loop 1 ; 
con < i n ; 
def z 32BIT ; 
z = + x y ; 
x = y ; 
y = z ; 
++ i ; 
end loop ; 
in 32BIT x ; 
call WRITE_UINT ; 
end ; `

**Comment by KittyCatOfMagic27** [comment]
VERSION 0.0.8 BETA RELEASED!!!!!
It is still under development, BUT I decided to release the half-baked version because why not! This new system compiles ADPL into ADIL _(the new intermediate language)_ and then that compiles to NASM. Because of this I can implement things later on that don't involve messing with the compilation of ADIL. This speeds up development and compilation _(or I assume it speeds up compilation comparing speeds to 0.0.7 and the fact that in ADIL compilation we are checking for less things, even with the janky/dead code in this version)_.

**Comment by KittyCatOfMagic27** [comment]
UPDATE TO VERSION 0.0.8 BETA!!!
You can check the README.md of the 0.0.8 BETA branch for updates in this version. Up next could be dynamic strings, arrays, linking, and a standard library! That's all, have a nice day. <3

**Comment by KittyCatOfMagic27** [comment]
Sorry for the lack of updates, I took a break for a bit.
The next version should be out in the next week or half a week at the most. I got caught up on adding pointer addition and subtaction as well as general bug fixing. Ifs and elses have been nearly fully implemeted, a way to make and traverse arrays has been added (all thats left is to make it more convient to use), and lots of native functions have been added. Some of the less nessary native functions and write functions will be moved to std.adpl once I figure out linking. 
TLDR: It'll be out soon, sorry for the lack of updates.
Have a great day!!! <3

**Comment by KittyCatOfMagic27** [comment]
UPDATING BETA
All that's left is some clean up of unneeded code and some bug fixing and I will finally be pushing 0.0.8!!
Onto some new documentation and then that'll be it for today, 
Have an AMAZING day!!! ...or else >:3

**Comment by KittyCatOfMagic27** [comment]
UPDATING BETA  FOR THE LAST TIME 'TIL NEW VER
Added elif back as else if, because it happened to be easier to implement. For now you can't put an else at the end of an else if chain and im aware of that bug. That should be fixed by release though and I have one more feature I'd like to add before the official release! And that feature issssssssssss... something you'll find out on release day :p 
That is it, I hope to all happy coding and to all a lovely night <3

**Comment by KittyCatOfMagic27** [comment]
NEW VERSION TOMORROW!!!!
I did everything that I said I would do in my last post, soooooooooo i have no idea what else to do until release. I may think about adding another feature on top of what I did, improve the documentation a bit (since there are some errors and things that are out of date), and/or some code clean up to help with future development and add room for more and potentially user made data types.  
That's all, have the BESTEST day ever!!! <3

**Comment by KittyCatOfMagic27** [comment]
VERSION 0.0.8 OUT NOW!!!!!!!!
